src/Microsoft.AspNetCore.Server.Kestrel/KestrelServer.cs

```diff
@@ -36,15 +34,19 @@ public KestrelServer(IFeatureCollection features, KestrelServerOptions options,
                 throw new ArgumentNullException(nameof(applicationLifetime));
             }
 
-            if (logger == null)
+            if (loggerFactory == null)
             {
-                throw new ArgumentNullException(nameof(logger));
+                throw new ArgumentNullException(nameof(loggerFactory));
             }
 
+            Options = options.Value ?? new KestrelServerOptions();
             _applicationLifetime = applicationLifetime;
-            _logger = logger;
-            Features = features;
-            Options = options;
+            _logger = loggerFactory.CreateLogger("Microsoft.AspNetCore.Server.Kestrel");
```

Comment:
I would use `typeof(KestrelServer).GetTypeInfo().Assembly.FullName` instead of hardcoding `"Microsoft.AspNetCore.Server.Kestrel"`.